### ISSUE TITLE
Evaluation mAP drops by 0.25 for new dataset.

### ISSUE THREAD

**Issue by s-arora-1987** [fizyr/keras-retinanet#1367]
Hi. I am trying to use keras retinanet for object detection in my experiments. Instead of focusing of multiple classes of object, I first verified it for only kind of object - a hand wearing blue glove. I have two annotated datasets: DS1 with around 3500 glove annotations, DS2 with 1500 annotations. I planned to train-validate-test this network on DS1 and do an additional evaluation on DS2. Testing is done using evaluate.py

I split DS1 in 80:10:10 for training:validation:testing. mAP result of evaluation on testing part of DS1 is around 0.97 However, mAP for testing on DS2 is around 0.72

I understand that the network was not trained on DS2, but location of recording video, background of video, and object's color are not changing in two datasets. Then, should evaluation mAP drop by that much for DS2 as compared to DS1?

**Comment by pleaseRedo** [comment]
0.97 a really high value, it could be overfitting. 
You don't tell us how DS1 and DS2 is formed. You could do followings, put ds1 ds2 together, shuffled and split to 3500 and 1500, this would guarantee no variation between dataset. Using the same weight train for few epochs and see how it's going then.

**Comment by hgaiser** [comment]
@pleaseRedo answered your question I believe. The two datasets might have minuscule differences, like you'd see from recording outside 2 hours apart. The shadows will be slightly different, which will impact results. If you want it to do well on situation X, then you will also have to train on X.
